src/osdc/Objecter.cc

```diff
@@ -2933,10 +2923,21 @@ void Objecter::_session_op_assign(OSDSession *to, Op *op)
   op->session = to;
   to->ops[op->tid] = op;
 
+  if(op->on_noosd_timeout && (! to->is_homeless())){
+    timer.cancel_event(op->on_noosd_timeout);
+    op->on_noosd_timeout = 0;
+    ldout(cct, 0) << __func__ << " session timeout cancelled" << dendl;
+  }
```

Comment:
it looks like `_session_op_remove()` would be a better place for this block, to build on the symmetry that already exists:
```c++
void Objecter::_session_op_assign(OSDSession *to, Op *op)
{
  ...
  if (to->is_homeless()) {
    num_homeless_ops++;
    // call timer.add_event()
  }
```
and
```c++
void Objecter::_session_op_remove(OSDSession *from, Op *op)
{
  ...
  if (from->is_homeless()) {
    num_homeless_ops--;
    // call timer.cancel_event()
  }
```

Reply:
Done